lib/serializer.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const Serializer = require('parse5/lib/serializer');
+
+module.exports = class CustomSerializer extends Serializer {
+    
+    constructor(node, options) {
+        super(node, options);
+        this.slotMap = options.slotMap;
+        this.handleTags = options.handleTags;
+        this.pipeTags = options.pipeTags;
+        this.isPipeInserted = false;
+        this.lastChildInserted = false;
+        this.serializedList = [];
+        this._serializeNode = this._serializeNode.bind(this);
+    }
+
+    pushBuffer() {
+        if (this.html !== '') {
+            this.serializedList.push(new Buffer(this.html));
+            this.html = '';
+        }
+    }
+
+    serialize() {
+        this._serializeChildNodes(this.startNode);
+        this.pushBuffer();
+        return this.serializedList;
+    }
+
+    _isPipeNode(node) {
+        return this.pipeTags.indexOf(node.name) !== -1;
+    }
+
+    _isSlotNode(node) {
+        if (node.name === 'slot') {
+            return true;
+        }
+        const attribs = node.attribs;
+        if (!attribs) {
+            return false;
+        }
+        return node.name === 'script' && attribs.type === 'slot';
+    }
+
+    _isSpecialNode(node) {
+        if (this.handleTags.indexOf(node.name) !== -1) {
+            return true;
+        }
+        const attribs = node.attribs;
+        if (attribs && attribs.type) {
+            return node.name === 'script' && this.handleTags.indexOf(attribs.type) !== -1;
+        }
+        return false;
+    }
+
+    _isLastChildOfBody(node) {
+        const parentNode = node.parent;
+        if (parentNode.name === 'body') {
+            if (Object.is(node, parentNode.lastChild)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    _serializeSpecial(node) {
+        this.pushBuffer();
+        let fragmentObj;
+        if (this.handleTags.indexOf(node.name) !== -1) {
```

Comment:
Not a necessary change but t probably would be more readable with "includes" from ES6